packages/cmd/src/services/cache/set.ts

```diff
@@ -82,6 +85,35 @@ export async function handleSetCache() {
   });
 
   success('Cache uploaded. Cache ID: %s', result.cacheId);
+
+  if (saveToHistory) {
+    if (uploadPaths.length > 0) {
+      await handleArchiveUpload({
+        archive: await zipFilesToBuffer(uploadPaths),
```

Comment:
Can we use the same buffer or a copy of the buffer to we aren't archiving twice?